samples/spring-security-xsuaa-usage/README.md

```diff
@@ -60,3 +60,65 @@ After deployment, the AppRouter will trigger authentication automatically when y
 * `https://spring-security-xsuaa-usage-web-<ID>.<LANDSCAPE_APPS_DOMAIN>/v2/sayHello` - GET request that provides generic Jwt info, but only if token matches.
 
 > Note: https://spring-security-xsuaa-usage-web-<ID>.<LANDSCAPE_APPS_DOMAIN> points to the url of the AppRouter. Get all app routes with `cf apps`.
+
+## Exchanging Tokens with `XsuaaTokenFlows`
+
+XSUAA introduces a new API to support the following token flows:
+
+* Authorization Code Grant
+* User Token Grant
+* Client Credentials Token Grant
+* Refresh Token Grant
+
+Authorization Code Grant is a flow that involves the browser and is therefore triggered by an API gateway (e.g. AppRouter). The other flows, however, may need to be triggered programmatically, e.g. to exchange one token for another or refresh a token, if it is about to expire.
+
+To that end, we now provide the `XsuaaTokenFlows` class, which serves as a factory for the different flows. This deprecates the existing `XsuaaToken.requestToken(XSTokenRequest)` API.
+
+The flows themselves provide a builder-pattern API that allows applications to easily create and execute each flow, guiding developers to only set properties that are relevant for the respective token flow.
+
+To consume the `XsuaaTokenFlows` class, you simply need to `@Autowire` it like this:
+```java
+@Autowired
+private XsuaaTokenFlows xsuaaTokenFlows;
+```
+
+Then, to create a **client credentials token flow** very easily by the following code:
+
+```java
+Jwt ccfToken = xsuaaTokenFlows.clientCredentialsTokenFlow(URI.create(baseUrl))
+                .client(clientId)
+                .secret(clientSecret)
+                .execute();
+```
+
+In case you have a refresh_token value and want to refresh an existing token with it, you can do the following:
+
+```java
+Jwt refreshToken = xsuaaTokenFlows.refreshTokenFlow(URI.create(baseUrl))
+                    .refreshToken("Your refresh token goes here. You get this from the OAuth server.")
```

Comment:
I considered both "dummy token" and an encoded / generated UUID string. 
Both options did not seem like good options for me, since they do not make clear where the token comes from. Also, a refresh token - afaik - has no standardized format. Its just a random value that is used by the server to make the match between a client, a token and a refreshed token. So, to me, this was the best option, also since it will fail at least with an error messag containing this "slightly more verbose" dummy token. 

Not sure why this confused you, though.